src/commands/join.cpp

```diff
@@ -22,103 +23,117 @@
+  stringVector::const_iterator it = channels.begin();
+  stringVector::const_iterator itEnd = channels.end();
+  for (; it != itEnd; ++it) {
+    if (it->empty() || (it->length() == 1 && (*it)[0] == REG_CHAN)) {
       send461NeedMoreParams(client, "JOIN");
-      continue;
+      return false;
     }
-
-    std::string channelNameWithoutPrefix = channelName.substr(1);
-
-    if (!isChannelValid(channelName, client)) {
-      continue;
+    if (it->find_first_of(" \t\n\r\f\v,") != std::string::npos) {
+      std::cout << RED
+          "ERROR : channel name cannot contain whitespaces or commas or be "
+          "empty" RESET
+                << std::endl;
+      return false;
     }
-    if (_channels.find(channelNameWithoutPrefix) == _channels.end()) {
-      createAndRegisterChannel(&client, channelNameWithoutPrefix);
-    }
-    const clientPMap &clientsInChannel =
-        _channels[channelName].getChannelClients();
-    if (clientsInChannel.find(fd) == clientsInChannel.end()) {
-      _channels.at(channelNameWithoutPrefix).addClientToChannelMap(&client);
-      client.incrementChannelsCount();
-      std::string key = (i < keys.size()) ? keys[i] : "";
-      Channel &channel = _channels[channelNameWithoutPrefix];
-
-      if (channel.getMode().keyRequired && key != channel.getKey()) {
-        send475BadChannelKey(client, channel);
-        continue;
-      }
-      handleJoinRequest(fd, client, channelNameWithoutPrefix);
+    if (it->length() > 20 || it->length() < 2 || (*it)[0] != REG_CHAN) {
+      send476BadChanMask(client, it->substr(1));
+      return false;
+    } else if (client.getChannelsCount() >= gConfig->getLimit("CHANLIMIT")) {
+      send405TooManyChannels(client);
+      return false;
     }
   }
+  return true;
 }
 
-bool Server::isValidChannelNameLength(const std::string &param) {
-  return param.length() >= 1 && param.length() < 51;
-}
-
-bool Server::isValidChannelPrefix(const std::string &param) {
-  return param[0] == REG_CHAN;
-}
-
-bool Server::isSingleCharacterChannelPrefix(const std::string &param) {
-  return param.length() == 1 && param[0] == REG_CHAN;
+void Server::checkKeysCorrectness(const stringVector &keys) {
+  stringVector::const_iterator it = keys.begin();
+  stringVector::const_iterator itEnd = keys.end();
+  for (; it != itEnd; ++it) {
+    if (it->find_first_of(" \t\n\r\f\v,") != std::string::npos) {
+      std::cout << RED
+          "ERROR : keys cannot contain whitespaces or commas or be empty" RESET
+                << std::endl;
+      return;
+    }
+  }
 }
 
-bool Server::isLeaveAllChannelsRequest(const std::string &param) {
-  return param == "0";
-}
+// tester avec Client &client
+void Server::processJoinRequest(int fd, Client *client,
+                                const std::string &channelName,
+                                const std::vector<std::string> &keys,
+                                size_t i) {
+  // si le channel n'existe pas on l'ajoute
```

Comment:
Comments should be written in English

Reply:
ces commentaires seront retires a la fin , cetait pour votre comprehension pour la PR